core/storage/story/gae_models_test.py

```diff
@@ -0,0 +1,61 @@
+# Copyright 2018 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests for Oppia story models."""
+import datetime
+
+from core.platform import models
+from core.tests import test_utils
+
+(story_models, base_models, ) = models.Registry.import_models(
+    [models.NAMES.story, models.NAMES.base_model])
+
+
+class StoryModelTest(test_utils.GenericTestBase):
+    """Tests for Oppia story models."""
+
+    def test_story_model(self):
+        committer_id = "test_committer_id"
+        commit_message = "test_commit_message"
+        commit_cmds = [{'cmd': 'test_command'}]
+
+        resp = story_models.StoryModel(
+            id='id', title='title',
+            description='description',
+            notes='notes',
+            language_code='language_code')
+        resp.commit(committer_id, commit_message, commit_cmds)
+        resp = resp.key.get()
+        self.assertEqual(resp.description, "description")
+        self.assertEqual(resp.title, 'title')
+
+
+class StorySummaryModelTest(test_utils.GenericTestBase):
+    """Tests for Oppia story summary models."""
+
+    def test_story_summary_model(self):
+
+        resp = story_models.StorySummaryModel(
+            id='id',
+            title='title',
+            description='description',
+            story_model_last_updated=datetime.datetime.utcnow(),
+            story_model_created_on=datetime.datetime.utcnow(),
+            language_code='language_code',
+            node_count=2,
+            version=1)
+        resp.put()
+        resp = resp.key.get()
```

Comment:
This is an interesting way to get an element by it's key. However, I think in most of the codebase, we use something similar to "story_models.StoryModel.get_by_id(story_id)". I would suggest using the get_by_id function as this is how the model gets queried in practice, and we need the test to assert that querying how we normally would should work fine without errors. 

Also, one thing to note is that, to get the instance, you are using the already existing instance (which isn't how code would work in practice).

(Please make a similar change in the previous test too)

Reply:
okay, let me fix that

Reply:
@nithusha21 the story model does not have "get_by_id". Do I need to first write the function in the models?

Reply:
Thanks alot...

Reply:
@nithusha21  I have made all the changes but just stuck on this comment you added at the end.
`Also, one thing to note is that, to get the instance, you are using the already existing instance (which isn't how code would work in practice).`
Help me elaborate on this so that i push my comit.

Reply:
Hi @JoyLubega, I was referring to the general situation that will occur on the server. The query will be either the id, or any of the parameters of the object (in this case the id). So, the target task is to fetch the object from the datastore given just the id of the object. So I was looking for something along the lines of get_by_id(object_id). 

In the earlier code where you did resp.key.get(), it was something like you know that resp is the object, you get its key and then get it again from the datastore. I was just saying that it would be better to do `story_models.StoryModel.get_by_id('id')`.

Reply:
Okay, thanks it clear now. Let me push the changes